cfy_manager/components/postgresql_server/postgresql_server.py

```diff
@@ -216,29 +233,174 @@ def _update_postgres_password(self):
+        systemd.enable('etcd', append_prefix=False)
+        self._start_etcd()
+        etcdctl_base_command = [
+            'etcdctl', '--endpoints', 'https://127.0.0.1:2379',
+            '--ca-file', ETCD_CA_PATH,
+        ]
+        cluster_state = common.run(
+            etcdctl_base_command + ['cluster-health'],
+            ignore_failures=True,
+        ).aggr_stdout
+        if 'cluster is degraded' in cluster_state:
+            cluster_config = config[POSTGRESQL_SERVER]['cluster']
+            # We want to configure etcd auth when the second node joins
+            # because that's the earliest we can do so.
+            logger.info('Configuring etcd authentication')
+            # Note that, per the etcd documentation, the root user must exist
+            # if authentication is to be enabled for etcd
+            common.run(
+                'echo {etcd_password} | {etcdctl} user add root'.format(
+                    etcd_password=cluster_config['etcd']['root_password'],
+                    etcdctl=' '.join(etcdctl_base_command)
+                ),
+                shell=True,
+            )
+            common.run(
+                'echo {etcd_password} | {etcdctl} user add patroni'.format(
+                    etcd_password=cluster_config['etcd']['patroni_password'],
+                    etcdctl=' '.join(etcdctl_base_command)
+                ),
+                shell=True,
+            )
+            common.run(
+                etcdctl_base_command +
+                ['user', 'grant', 'patroni', '--roles', 'guest']
+            )
+            common.run(etcdctl_base_command + ['auth', 'enable'])
+
+        logger.info('Creating postgres bin links for patroni')
+        for pg_bin in PG_BINS:
+            common.sudo(['ln', '-s', os.path.join(PG_BIN_DIR, pg_bin),
+                         '/usr/sbin'])
+
+        logger.info('Starting patroni')
+        files.deploy(
+            os.path.join(CONFIG_PATH, 'patroni.service'),
+            '/usr/lib/systemd/system/patroni.service',
+            render=False,
+        )
+        systemd.enable('patroni', append_prefix=False)
+        systemd.start('patroni', append_prefix=False)
 
     def install(self):
         logger.notice('Installing PostgreSQL Server...')
-        self._install()
+        sources = config[POSTGRESQL_SERVER][SOURCES]
+
+        logger.debug('Installing PostgreSQL Server dependencies...')
+        yum_install(sources['libxslt_rpm_url'])
+
+        logger.debug('Installing PostgreSQL Server...')
+        yum_install(sources['ps_libs_rpm_url'])
+        yum_install(sources['ps_rpm_url'])
+        yum_install(sources['ps_contrib_rpm_url'])
+        yum_install(sources['ps_server_rpm_url'])
+        yum_install(sources['ps_devel_rpm_url'])
+        # As we don't support installing community as anything other than AIO,
+        # not having manager service installed means that this must be premium
+        if MANAGER_SERVICE not in config[SERVICES_TO_INSTALL]:
```

Comment:
i'd put this check to a separate `self._is_only_database()` method and make the comment be a docstring

Reply:
It's not for only database though- I don't know whether we want to support installing rabbit with it, but if rabbit is colocated then it won't make any difference to this installer.